File: apps/api-dotnet/src/AdventureWorks.Models/Features/HumanResources/EmployeeOrgTreeItemModel.cs
namespace AdventureWorks.Models.Features.HumanResources;

public sealed class EmployeeOrgTreeItemModel
{
    public int EmployeeId { get; set; }
    public required string FullName { get; set; }
    public required string JobTitle { get; set; }
    public required string DepartmentName { get; set; }
    public short? OrganizationLevel { get; set; }
    public int? ParentEmployeeId { get; set; }
}

File: apps/angular-web/libs/hr/data-access/src/lib/models/employee-org-tree-item.model.ts
/** Flat org-hierarchy row from GET /v1/employees/org-tree. The root (CEO) has organizationLevel and parentEmployeeId both null. */
export interface EmployeeOrgTreeItem {
  readonly employeeId: number;
  readonly fullName: string;
  readonly jobTitle: string;
  readonly departmentName: string;
  readonly organizationLevel: number | null;
  readonly parentEmployeeId: number | null;
}
